src/Microsoft.AspNetCore.TestHost/ResponseStream.cs

```diff
@@ -266,24 +253,7 @@ public override void Write(byte[] buffer, int offset, int count)
                 _writeLock.Release();
             }
         }
-#if NET451
-        public override IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback callback, object state)
```

Comment:
We ran into deadlocks in Kestrel when not implementing these methods on full framework. Not sure if that applies here though.